src/modules/types.rs

```diff
@@ -30,6 +30,14 @@ impl Type {
     pub fn is_allowed_in(&self, other: &Type) -> bool {
         self == other || self.is_subset_of(other)
     }
+
+    pub fn is_array(&self) -> bool {
```

Comment:
This may cause problems if the union types change supports union types in *return values*.  If a function could return `Text | [Text]` (which seems like something we would want to disallow, tbh) what would the default return type be?